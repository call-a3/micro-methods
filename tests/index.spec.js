import withMethods from '../src/index'

describe('micro-methods', () => {
  it('should serve a request with a supported method', async () => {
    const GET = jest.fn()
    const decorated = withMethods({ GET })

    const res = {
      setHeader: jest.fn(),
      getHeader: jest.fn(() => true),
      end: jest.fn(),
    }
    await expect(decorated({ method: 'GET' }, res)).resolves.toEqual(undefined)
  })
  it('should fail to serve a request with an unsupported method', async () => {
    const GET = jest.fn()
    const decorated = withMethods({ GET })

    const res = {
      setHeader: jest.fn(),
      getHeader: jest.fn(() => true),
      end: jest.fn(),
    }
    await expect(decorated({ method: 'POST' }, res)).resolves.toEqual(undefined)
    expect(res).toHaveProperty('statusCode', 405)
    expect(res.setHeader).toHaveBeenCalled()
    expect(res.setHeader.mock.calls).toMatchSnapshot('setHeader calls')
    expect(res.end).toHaveBeenCalledTimes(1)
    expect(res.end.mock.calls[0]).toMatchSnapshot('body')
  })
})
