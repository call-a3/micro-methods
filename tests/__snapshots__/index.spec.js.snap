// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`micro-methods should fail to serve a request with an unsupported method: body 1`] = `
Array [
  "{\\"title\\":\\"Method Not Allowed\\",\\"status\\":405,\\"detail\\":\\"The HTTP method you specified is not supported by the requested resource. Check if you should be using one of the supported methods instead.\\",\\"supported\\":[\\"GET\\"],\\"requested\\":\\"POST\\"}",
]
`;

exports[`micro-methods should fail to serve a request with an unsupported method: setHeader calls 1`] = `
Array [
  Array [
    "Content-Type",
    "application/json+problem",
  ],
  Array [
    "Content-Length",
    234,
  ],
]
`;
